src/tests/_internal/core/models/test_configurations.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any, Optional
+
+import pytest
+
+from dstack._internal.core.errors import ConfigurationError
+from dstack._internal.core.models.configurations import parse
+from dstack._internal.core.models.resources import Range
+
+
+class TestParseConfiguration:
+    def test_services_replicas_and_scaling(self):
+        def test_conf(replicas: Any, scaling: Optional[Any] = None):
+            conf = {
+                "type": "service",
+                "commands": ["python3 -m http.server"],
+                "port": 8000,
+                "replicas": replicas,
+            }
+            if scaling:
+                conf["scaling"] = scaling
+            return conf
+
+        assert parse(test_conf(1)).replicas == Range(min=1, max=1)
+        assert parse(test_conf("2")).replicas == Range(min=2, max=2)
+        assert parse(test_conf("3..3")).replicas == Range(min=3, max=3)
+        with pytest.raises(
+            ConfigurationError,
+            match="When you set `replicas` to a range, ensure to specify `scaling`",
+        ):
+            parse(test_conf("0..10"))
+        assert parse(
+            test_conf(
+                "0..10",
+                {
+                    "metric": "rps",
+                    "target": 10,
+                },
+            )
+        ).replicas == Range(min=0, max=10)
+        with pytest.raises(
+            ConfigurationError,
+            match="When you set `replicas` to a range, ensure to specify `scaling`",
```

Comment:
(nit) I think the primary error here is `unexpected value; permitted: 'rps'`. The `ensure to specify scaling` error also appears, but as a consequence of the primary error